README.md

```diff
@@ -99,106 +107,132 @@ evaluator = MulticlassClassificationEvaluator(metricName="accuracy")
-works on TensorFlow-backed Keras models. It
-* Internally creates a DataFrame containing a column of images by applying the user-specified image
-loading and processing function to the input DataFrame containing a column of image URIs
-* Loads a Keras model from the given model file path
-* Applies the model to the image DataFrame
+3.  For Keras users
 
-The difference in the API from `TFImageTransformer` above stems from the fact that usual Keras
-workflows have very specific ways to load and resize images that are not part of the TensorFlow Graph.
+    For applying Keras models in a distributed manner using Spark, `KerasImageFileTransformer` works on TensorFlow-backed Keras models. It
 
+    1.  Internally creates a DataFrame containing a column of images by applying the user-specified image loading and processing function to the input DataFrame containing a column of image URIs
+    2.  Loads a Keras model from the given model file path
+    3.  Applies the model to the image DataFrame
 
-To use the transformer, we first need to have a Keras model stored as a file. For this example we'll 
-just save the Keras built-in InceptionV3 model instead of training one.
+    The difference in the API from `TFImageTransformer` above stems from the fact that usual Keras workflows have very specific ways to load and resize images that are not part of the TensorFlow Graph.
+
+    To use the transformer, we first need to have a Keras model stored as a file. For this example we'll just save the Keras built-in InceptionV3 model instead of training one.
+
+    ```python
+    from keras.applications import InceptionV3
+
+    model = InceptionV3(weights="imagenet")
+    model.save('/tmp/model-full.h5')
+    ```
+
+    Now on the prediction side, we can do:
+
+    ```python
+    from keras.applications.inception_v3 import preprocess_input
+    from keras.preprocessing.image import img_to_array, load_img
+    import numpy as np
+    import os
+    from sparkdl import KerasImageFileTransformer
+
+    def loadAndPreprocessKerasInceptionV3(uri):
+        # this is a typical way to load and prep images in keras
+        image = img_to_array(load_img(uri, target_size=(299, 299)))
+        image = np.expand_dims(image, axis=0)
+        return preprocess_input(image)
+
+    transformer = KerasImageFileTransformer(inputCol="uri", outputCol="predictions",
+                                            modelFile="/tmp/model-full.h5",
+                                            imageLoader=loadAndPreprocessKerasInceptionV3,
+                                            outputMode="vector")
+
+    files = [os.path.abspath(os.path.join(dirpath, f)) for f in os.listdir("/data/myimages") if f.endswith('.jpg')]
+    uri_df = sqlContext.createDataFrame(files, StringType()).toDF("uri")
+
+    final_df = transformer.transform(uri_df)
+    ```
+
+
+### Deploying models as SQL functions
+
```

Comment:
Let's add some more context:

"One way to productionize a model is to deploy it as a Spark SQL function, which allows anyone who knows SQL to use it. Deep Learning Pipelines provides mechanisms to take a deep learning model and register a Spark SQL user-defined function (UDF). "